src/operator/nn/lrn.cc

```diff
@@ -42,6 +50,22 @@ static bool LRNShape(const nnvm::NodeAttrs& attrs,
   out_shape->clear();
   out_shape->push_back(dshape);
   out_shape->push_back(dshape);
+#if MXNET_USE_MKLDNN == 1
+  // Create LRN primitive for getting the workspace size
+  CHECK_EQ(dshape.ndim(), 4U);
```

Comment:
In general, MKL-DNN is specific for deep learning so that the default input tensor is 4D for all OPs. Currently, the 4D shape is fully supported by MKLDNN and 2D is also can work for OPs.
http://01org.github.io/mkl-dnn/group__c__api__lrn.html

And 2D shape in here is not very computation intensive so it will introduce extra overhead by using MKL-DNN. I suggest to taking over 4D calculation in here.